rust/module.rs

```diff
@@ -171,7 +170,14 @@ fn __build_modinfo_string_base(
 }
 
 fn __build_modinfo_string_variable(module: &str, field: &str) -> String {
-    format!("__{module}_{field}", module = module, field = field)
+    static COUNTER: AtomicUsize = AtomicUsize::new(0);
```

Comment:
This seemed a little heavy handed, but currently `__MODULE_INFO` uses `__UNIQUE_id` which adds in `__COUNTER__` to the variable name. This was the simplest quick thing I could think of. Without the added count we try to reuse the same name for each softdep.

Reply:
Could this function take `counter: &mut usize` as argument and then pass this around such that each `module!()` invocation uses the same `counter`. You can then increment this counter and use it. Proc macros are ideally pure functions of their arguments. The order of invocation of proc macros is not guaranteed in case of rustc and proc macros may be invoked multiple times in case of rust-analyzer. Rustc doesn't care about non-determinism in proc macros at the moment, but rust-analyzer used to crash on non-deterministic proc macros. This has since been fixed, but it is still not a good idea.

Reply:
Updated to use a `usize` here.